elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -157,11 +157,7 @@ export abstract class BaseTransaction {
 	}
 
 	public get id(): string {
-		if (!this._id) {
-			throw new Error('id is required to be set before use');
-		}
-
-		return this._id;
+		return this._id || 'incalculable-id';
```

Comment:
why not empty string?

Reply:
to be more obvious to the user that something is wrong with its JSON